packages/react-popup/Spec.md

```diff
@@ -2,30 +2,375 @@
+
+```tsx
+// Uncontrolled
+<Popup
+  trigger={<Button icon={<MoreIcon />} title="Show popup" />}
+  content={popupContent}
+  on=['click', 'hover', 'context', 'focus']
+/>
+
+// Controlled
+<Popup
+  open={open}
+  onOpenChange={(e, { open }) => setOpen(open)}
+  trigger={<Button icon={<MoreIcon />} title="Show popup" />}
+  content={popupContent}
+  on=['click', 'hover', 'context', 'focus']
+/>
+```
+
+Focus trap is enabled using the `trapFocus` prop.
+
+A `target` prop also exists so that the `Popup` does not have to be anchored to the `trigger` element.
+
+### Position/Alignment hints
+
+Both libraries provide an API that achieves the same end result for positioning and alignment. Below is a table that maps the v8 `DirectionalHint` with the v0 props of `position` and `alignment`
+
+| DirectionalHint (v7) | position (v0) | align (v0) |
+| -------------------- | ------------- | ---------- |
+| topLeftEdge          | above         | start      |
+| topCenter            | above         | center     |
+| topRightEdge         | above         | bottom     |
+| topAutoEdge          | above         |            |
+| bottomLeftEdge       | above         | start      |
+| bottomCenter         | below         | center     |
+| bottomRightEdge      | below         | bottom     |
+| bottomAutoEdge       | below         |            |
+| leftTopEdge          | before        | top        |
+| leftCenter           | before        | center     |
+| leftBottomEdge       | before        | bottom     |
+| rightTopEdge         | after         | before     |
+| rightCenter          | after         | center     |
+| rightBottomEdge      | after         | bottom     |
+
+v8 uses `left` and `right`. v0 uses `before` and `after`. v0 vocabulary tries to be consistent regardless of RTL state. It's also possible to supply an explicit RTL hint to v8 which is a flip by default. v0 will flip by default but requires the consumer to detect RTL scenarios and modify props in these situations
+
+In general the separation of both the position and alignment in v0 results in an API that is easier to use if a consumer only needs to modify one of the two props. However both try to achieve the same result in the end.
+
+It's important to note that if an incorrect pair of `position` and `align` are provided in v0, then `position` takes priority and `align` is set to `center`
+
+### Offset
+
+```typescript
+<ContextualMenu
+  // single number value
+  gap={100}
+/>
+
+<Popup
+  offset={[-100, 100]}
+/>
+
+// offset can also be a function of raw Popper properties
+const offsetFunction = ({
+  popper: PopperJs.Rect;
+  reference: PopperJs.Rect;
+  placement: PopperJs.Placement;
+}) => ([popper.width, -popper.height])
+```
+
+v8 positioning can only apply a numerical value to the first part position attribute of `DirectionalHint`. v0 supports a function to defer calculation at runtime. v0 Also supports offset of the `Popup` in both axes.
```

Comment:
I think that it's important to mention that `offset` also handles RTL i.e values will be flipped when we are in RTL layout.